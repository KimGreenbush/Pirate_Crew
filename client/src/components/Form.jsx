import React, { useState } from "react"

const Form = ({initialName, onSubmitProp, errors}) => {
    const [name, setName] = useState(initialName)
    const [image, setImage] = useState("")
    const [chests, setChests] = useState(0)
    const [phrase, setPhrase] = useState("")
    const [position, setPosition] = useState("Captain")
    const [peg, setPeg] = useState(true)
    const [patch, setPatch] = useState(true)
    const [hook, setHook] = useState(true)

    const positions = ["Captain", "First Mate", "Quarter Master", "Boatswain", "Powder Monkey"]

    const submitHandler = e => {
        e.preventDefault()
        onSubmitProp({ name, image, chests, phrase, position, peg, patch, hook })
    }

    return (
        <div  className="pirate-form">
            { errors? errors.map((error, idx) => <p key={idx}>{error}</p>) : ""}
            <form onSubmit={submitHandler}>
                <label htmlFor="name">Pirate Name: </label>
                <input type="text" name="name" id="name" value={name} onChange={e => setName(e.target.value)} />
                <label htmlFor="image">Image Url: </label>
                <input type="text" name="image" id="image" value={image} onChange={e => setImage(e.target.value)} />
                <label htmlFor="chests"># of Treasure Chests:</label>
                <input type="number" name="chests" id="chests" value={chests} onChange={e => setChests(e.target.value)} />
                <label htmlFor="phrase">Pirate Catch Phrase:</label>
                <textarea value={phrase} onChange={e => setPhrase(e.target.value)} />
                <label htmlFor="position">Crew Position:</label>
                <select value={position} onChange={e => setPosition(e.target.value)}>
                    {positions.map((pos, idx) => <option key={idx} value={pos}>{pos}</option>)}
                </select>
                <label htmlFor="peg">Peg Leg</label>
                <input type="checkbox" checked={peg} onChange={e => setPeg(e.target.checked)}/>
                <label htmlFor="peg">Eye Patch</label>
                <input type="checkbox" checked={patch} onChange={e => setPatch(e.target.checked)}/>
                <label htmlFor="peg">Hook Hand</label>
                <input type="checkbox" checked={hook} onChange={e => setHook(e.target.checked)}/>
                <button>Add Pirate</button>
            </form>
        </div>
    )
}

export default Form