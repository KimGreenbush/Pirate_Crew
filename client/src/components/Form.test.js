/**
 * @jest-environment jsdom
 */

import React from "react";
import Form from "./Form";
import renderer from "react-test-renderer";

test("renders Form without crashing", () => {
	const component = renderer.create(<Form />);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});
