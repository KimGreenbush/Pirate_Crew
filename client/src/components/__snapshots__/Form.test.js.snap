// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Form without crashing 1`] = `
<div
  className="pirate-form"
>
  
  <form
    onSubmit={[Function]}
  >
    <label
      htmlFor="name"
    >
      Pirate Name: 
    </label>
    <input
      id="name"
      name="name"
      onChange={[Function]}
      type="text"
    />
    <label
      htmlFor="image"
    >
      Image Url: 
    </label>
    <input
      id="image"
      name="image"
      onChange={[Function]}
      type="text"
      value=""
    />
    <label
      htmlFor="chests"
    >
      # of Treasure Chests:
    </label>
    <input
      id="chests"
      name="chests"
      onChange={[Function]}
      type="number"
      value={0}
    />
    <label
      htmlFor="phrase"
    >
      Pirate Catch Phrase:
    </label>
    <textarea
      onChange={[Function]}
      value=""
    />
    <label
      htmlFor="position"
    >
      Crew Position:
    </label>
    <select
      onChange={[Function]}
      value="Captain"
    >
      <option
        value="Captain"
      >
        Captain
      </option>
      <option
        value="First Mate"
      >
        First Mate
      </option>
      <option
        value="Quarter Master"
      >
        Quarter Master
      </option>
      <option
        value="Boatswain"
      >
        Boatswain
      </option>
      <option
        value="Powder Monkey"
      >
        Powder Monkey
      </option>
    </select>
    <label
      htmlFor="peg"
    >
      Peg Leg
    </label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="peg"
    >
      Eye Patch
    </label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    <label
      htmlFor="peg"
    >
      Hook Hand
    </label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
    <button>
      Add Pirate
    </button>
  </form>
</div>
`;
